pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -279,11 +292,43 @@ func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error
 			Reason:  remoteFinishedCond.Reason,
 			Message: remoteFinishedCond.Message,
 		})
-		return a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
+		return reconcile.Result{}, a.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
 	}
 
 	hasReserving, reservingRemote := group.FirstReserving()
 
+	// if the reserving workload is out of sync, remove it
+	if hasReserving {
+		outOfSync := group.local == nil || !equality.Semantic.DeepEqual(group.local.Spec, group.remotes[reservingRemote].Spec)
+		if outOfSync {
+			if err := client.IgnoreNotFound(group.RemoveRemoteObjects(ctx, reservingRemote)); err != nil {
+				log.V(2).Error(err, "Deleting out of sync remote objects", "remote", reservingRemote)
+				return reconcile.Result{}, err
+			}
+			group.remotes[reservingRemote] = nil
+
+			// check if another one exists
+			hasReserving, reservingRemote = group.FirstReserving()
+		}
+	}
+
+	// If there is no reserving and the AC is ready, the connection with the reserving remote might
+	// be lost, keep the workload admitted for keepReadyTimeout and put it back in the queue after that.
+	if !hasReserving && acs.State == kueue.CheckStateReady {
+		remainingWaitTime := a.workerLostTimeout - time.Since(acs.LastTransitionTime.Time)
+		if remainingWaitTime > 0 {
+			log.V(3).Info("Reserving remote lost, retry", "retryAfter", remainingWaitTime)
+			return reconcile.Result{RequeueAfter: remainingWaitTime}, nil
```

Comment:
Requeue after 15min to check sounds bad. I think by analogy to `--retry` in curl we could have exponential backoff, or fixed delay (probably at 30s by default). Then wait for `min(remainingTime, delay)`

Reply:
The 15 min will apply only if no other event triggers the reconcile, then and only then we are interested in requeuing if the reserving worker is not back.

Reply:
Maybe I still miss some piece of information. Some questions: if we lose connectivity for 5min, do we mark the cluster a Say we have a workload which is running on worker1. Then, worker1 is marked as inactive (the AdmissionCheck is PartiallyActive). What will trigger the check to include the cluster back as active?

Also, if we don't get connectivity for 15min, do we mark such cluster as inactive?

Reply:
> Maybe I still miss some piece of information. Some question: if we have a workload running on worker1, and we lose connectivity for 5min, to worker1 (and worker1 is marked inactive). What will trigger the check to include the cluster back as active?

Nothing for now we have a follow-up for connection monitoring and reconnects.

> Here we are talking about workloads.

Reply:
Ok, I see, but there is an issue with delaying the check, that if we get unlucky once and 15min later, then re requeue. However, the cluster might have been available in-between for 14min. I don't think this is the intention. I think we should monitor if the cluster is connected, and switch its status based on that. 

If the problem is solved under " connection monitoring and reconnects.", then maybe it makes sense to combine to see the solution to temporary loss of connectivity holistically. At least update the "special notes to reviewers" with what will be covered in the coming follow up.

Reply:
When the watch starts, events are generated for the already exiting objects, hence at that point in time when the worker is alive for some time we wold get a reconcile also.

Reply:
But we still would not observe the cluster being available for 10min in between, This seems fragile to me. I think it is better to periodically check the status, then if the ping fails, we count 15min since the last successful status check.

Reply:
This is the best I can do.

Reply:
You mean in this PR / approach on in general we should not track the availability of a worker?

Reply:
In this PR, we should talk about this PR.